plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/conditions/accessors/HardcodedPropertyAccessorRegistry.java

```diff
@@ -0,0 +1,131 @@
+ */
+package org.apache.unomi.plugins.baseplugin.conditions.accessors;
+
+import org.apache.unomi.api.*;
+import org.apache.unomi.api.campaigns.Campaign;
+import org.apache.unomi.api.goals.Goal;
+import org.apache.unomi.api.rules.Rule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * This class contains the registry of all the hardcoded property accessors.
+ * For the moment this list of accessors is hardcoded, but in a future update it could be made dynamic.
+ */
+public class HardcodedPropertyAccessorRegistry {
+
+    private static final Logger logger = LoggerFactory.getLogger(HardcodedPropertyAccessorRegistry.class.getName());
+
+    Map<String, HardcodedPropertyAccessor> propertyAccessors = new HashMap<>();
+
+    public HardcodedPropertyAccessorRegistry() {
+        propertyAccessors.put(Item.class.getName(), new ItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(MetadataItem.class.getName(), new MetadataItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Metadata.class.getName(), new MetadataHardcodedPropertyAccessor(this));
+        propertyAccessors.put(TimestampedItem.class.getName(), new TimestampedItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Event.class.getName(), new EventHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Profile.class.getName(), new ProfileHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Consent.class.getName(), new ConsentHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Session.class.getName(), new SessionHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Rule.class.getName(), new RuleHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Goal.class.getName(), new GoalHardcodedPropertyAccessor(this));
+        propertyAccessors.put(CustomItem.class.getName(), new CustomItemHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Campaign.class.getName(), new CampaignHardcodedPropertyAccessor(this));
+        propertyAccessors.put(Map.class.getName(), new MapHardcodedPropertyAccessor(this));
+    }
+
+    public static class NextTokens {
+        public String propertyName;
+        public String leftoverExpression;
+    }
+
+    protected NextTokens getNextTokens(String expression) {
+        NextTokens nextTokens = new NextTokens();
+        if (expression.startsWith("[\"")) {
+            int lookupNameBeginPos = "[\"".length();
+            int lookupNameEndPos = expression.indexOf("\"].", lookupNameBeginPos);
+            if (lookupNameEndPos > lookupNameBeginPos) {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos, lookupNameEndPos);
+                nextTokens.leftoverExpression = expression.substring(lookupNameEndPos+2);
+            } else {
+                nextTokens.propertyName = expression.substring(lookupNameBeginPos);
+                nextTokens.leftoverExpression = null;
+            }
```

Comment:
`["foo"].bar` is covered
`["foo"]` is not covered

Also could be great to cover this function with unit test, since it's covering a lot of cases, and it should not be that complicated todo.